doc/services/ipc/ipc_service/ipc_service.rst

```diff
@@ -0,0 +1,252 @@
+.. _ipc_service:
+
+IPC service
+###########
+
+.. contents::
+    :local:
+    :depth: 2
+
+The IPC service API provides an interface to exchange the data between two
+domains/CPUs. Messages are exchanged between the domains through a shared
+memory region. Domains signal to each other data has been sent (or received)
+with use of IPM/MBOX devices.
+
```

Comment:
```
Messages are exchanged between the domains through a shared
memory region. Domains signal to each other data has been sent (or received)
with use of IPM/MBOX devices
```
This seems to me to be an assumption about the implementation of the backend.
For instance I don't see a limitation to use the IPC service for communication between two SoCs using an UART backend.

Perhaps be more generic at this level speaking about mechanism to exchange data and to signaling to inform about the message transfer.

Reply:
Afaik that was the intention of ipc service in the first place. It supports communication based on shared memory. Isn't that the case @carlocaione ?
I understand your point and i agree that technically there is a possibility to add uart backend.

Reply:
@emob-nordic can you make that statement a bit more generic to accommodate for other implementations that do not use IPM/MBOX as well? I kinda agree with @arnopo here

Reply:
@carlocaione yes, sure. I agree with that point. Thanks for noticing @arnopo

Reply:
What do you think about:
```
The IPC service API provides an interface to exchange the data between two
domains/CPUs. The data exchange is signaled between the domains after they are
sent.
```

Reply:
I would just remove the `The data exchange is signaled between the domains after they are
sent.` part. If that's done then the backend part would specify that.

Reply:
Sorry i removed the comment as i found what i wrote was silly. I pushed a change - i hope it's generic enough.